### PR TITLE
Don't crash on xatlas degenerate faces.

### PR DESCRIPTION
**Pull request by fire**
Don't crash on xatlas degenerate faces.

Sponsored by IMVU, Inc.

### CODE REVIEW

File: thirdparty/xatlas/xatlas.cpp [hunk 1]
```diff
@@ -1955,7 +1955,10 @@ class Mesh
 
 		//make sure its not a degenerate
 		bool degenerate = distanceSquared(base[0], base[1]) < NV_EPSILON || distanceSquared(base[0], base[2]) < NV_EPSILON || distanceSquared(base[1], base[2]) < NV_EPSILON;
-		xaDebugAssert(!degenerate);
+		if(degenerate) {
```
Comment (YeldhamDev) [inline thirdparty/xatlas/xatlas.cpp]: There should be a space between the "if" and the parenthesis.

Comment (fire) [reply thirdparty/xatlas/xatlas.cpp]: Resolved. Thanks.

File: thirdparty/xatlas/xatlas.cpp [hunk 2]
```diff
@@ -1955,7 +1955,10 @@ class Mesh
 
 		//make sure its not a degenerate
 		bool degenerate = distanceSquared(base[0], base[1]) < NV_EPSILON || distanceSquared(base[0], base[2]) < NV_EPSILON || distanceSquared(base[1], base[2]) < NV_EPSILON;
-		xaDebugAssert(!degenerate);
+		if (degenerate) {
```
Comment (akien-mga) [inline thirdparty/xatlas/xatlas.cpp]: Please mark those Godot-specifix changes enclosed with `// -- GODOT start --` and `//-- GODOT end --`

Comment (akien-mga) [reply thirdparty/xatlas/xatlas.cpp]: And include a .patch file with this change or merge in the other related patch.

Comment (akien-mga) [reply thirdparty/xatlas/xatlas.cpp]: You can do e.g.:
```
cd thirdparty/xatlas/
git apply -R avoid-failing-on-bad-geometry.patch
git add xatlas.cpp 
git apply avoid-failing-on-bad-geometry.patch
// do your own changes
git diff xatlas.cpp > avoid-failing-on-bad-geometry.patch
git add xatlas.cpp
```
That should update the patch and leave your additional changes to commit.

Comment (fire) [reply thirdparty/xatlas/xatlas.cpp]: Will look into this this weekend.

Comment (fire) [reply thirdparty/xatlas/xatlas.cpp]: Instead of revising this patch, xatlas will be upgraded to the latest version.

**Comment by akien-mga** [comment]
> Instead of revising this patch, xatlas will be upgraded to the latest version.

Closing for now then, as upgrading xatlas and working with upstream to solve remaining issues is definitely the better approach.

If this isn't possible for some reason, we can reopen this PR to rework this patch.

**Pull request closed without merging**